File: LeetCode/1266_MinimumTimeVisitingAllPoints/lc1266.cpp
#include <cassert>
#include <cmath>
#include <vector>

class Solution {
public:
  int minTimeToVisitAllPoints(std::vector<std::vector<int>>& points) {
    int ans = 0;
    for (int i = 1; i < points.size(); ++i) {
      ans += std::max(std::abs(points[i][0] - points[i-1][0]), std::abs(points[i][1] - points[i-1][1]));
    }
    return ans;
  }
};

int main() {
  Solution s;
  {
    std::vector<std::vector<int>> points = {{1, 1}, {3, 4}, {-1, 0}};
    assert(s.minTimeToVisitAllPoints(points) == 7);
  }
  {
    std::vector<std::vector<int>> points = {{3, 2}, {-2, 2}};
    assert(s.minTimeToVisitAllPoints(points) == 5);
  }
}

File: LeetCode/1266/golang/lc1266.go
package lc1266

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

// Runtime complexity: O(n)
// Auxiliary space: O(1)
// Subjective level: easy
// Solved on: 2026-01-12
func minTimeToVisitAllPoints(points [][]int) int {
	n := len(points)
	ans := 0
	for i := 1; i < n; i++ {
		ans += max(abs(points[i][0]-points[i-1][0]), abs(points[i][1]-points[i-1][1]))
	}
	return ans
}
